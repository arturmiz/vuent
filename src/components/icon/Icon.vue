<script>
import VntIconContact from './icons/Contact.vue';
import VntIconEdit from './icons/Edit.vue';
import VntIconGlobalNav from './icons/GlobalNavigation.vue';
import VntIconSave from './icons/Save.vue';
import VntIconSettings from './icons/Settings.vue';

export default {
  name: 'VntIcon',

  functional: true,

  components: {
    VntIconContact,
    VntIconEdit,
    VntIconGlobalNav,
    VntIconSave,
    VntIconSettings
  },

  props: {
    name: {
      type: String,
      default: undefined
    }
  },

  render(createElement, { props, data }) {
    function resolveIcon(name) {
      switch (name) {
      case 'contact':
        return VntIconContact;
      case 'edit':
        return VntIconEdit;
      case 'global-nav':
      case 'global-navigation':
        return VntIconGlobalNav;
      case 'save':
        return VntIconSave;
      case 'settings':
        return VntIconSettings;
      default:
        return null;
      }
    }

    return createElement('span', {
      class: data.staticClass ? `vnt-icon ${data.staticClass}` : 'vnt-icon',
      attrs: data.attrs
    }, [
      createElement(resolveIcon(props.name))
    ]);
  }
};
</script>

<style lang="scss">
.vnt-icon {
  display: inline-block;
  width: 16px;
  height: 16px;
}
</style>
