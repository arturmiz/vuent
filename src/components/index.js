import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
