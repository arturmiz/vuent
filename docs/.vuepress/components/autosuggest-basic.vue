<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
