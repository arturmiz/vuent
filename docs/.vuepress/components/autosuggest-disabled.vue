<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
