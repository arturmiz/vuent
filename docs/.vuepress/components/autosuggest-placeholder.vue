<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
