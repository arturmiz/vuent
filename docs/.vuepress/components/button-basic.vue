<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
