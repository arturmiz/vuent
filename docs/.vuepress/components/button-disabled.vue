<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
