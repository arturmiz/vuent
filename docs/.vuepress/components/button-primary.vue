<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
