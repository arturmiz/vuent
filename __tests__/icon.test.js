import { createLocalVue, mount } from '@vue/test-utils';
import { VntIcon } from '@/components';
import { isInstalled } from './utils';
import { resolveIcon } from '@/components/icon/Icon.vue';

import VntIconAdd from '@/components/icon/icons/Add.vue';
import VntIconClose from '@/components/icon/icons/Close.vue';
import VntIconContact from '@/components/icon/icons/Contact.vue';
import VntIconDelete from '@/components/icon/icons/Delete.vue';
import VntIconEdit from '@/components/icon/icons/Edit.vue';
import VntIconFlag from '@/components/icon/icons/Flag.vue';
import VntIconGlobalNav from '@/components/icon/icons/GlobalNavigation.vue';
import VntIconHome from '@/components/icon/icons/Home.vue';
import VntIconSettings from '@/components/icon/icons/Settings.vue';
import VntIconSave from '@/components/icon/icons/Save.vue';
import VntIconTag from '@/components/icon/icons/Tag.vue';

describe('Icon', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntIcon);

    expect(isInstalled(localVue, VntIcon)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntIcon);
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('given icon\'s key', () => {

    beforeAll(() => {
      wrapper = mount(VntIcon, {
        propsData: {
          name: 'settings'
        }
      });
    });

    test('uses correct component', () => {
      expect(resolveIcon('add').name).toBe(VntIconAdd.name);
      expect(resolveIcon('close').name).toBe(VntIconClose.name);
      expect(resolveIcon('contact').name).toBe(VntIconContact.name);
      expect(resolveIcon('delete').name).toBe(VntIconDelete.name);
      expect(resolveIcon('edit').name).toBe(VntIconEdit.name);
      expect(resolveIcon('flag').name).toBe(VntIconFlag.name);
      expect(resolveIcon('global-navigation').name).toBe(VntIconGlobalNav.name);
      expect(resolveIcon('global-nav').name).toBe(VntIconGlobalNav.name);
      expect(resolveIcon('home').name).toBe(VntIconHome.name);
      expect(resolveIcon('save').name).toBe(VntIconSave.name);
      expect(resolveIcon('settings').name).toBe(VntIconSettings.name);
      expect(resolveIcon('tag').name).toBe(VntIconTag.name);
      expect(resolveIcon('some-non-existing-icon')).toBeNull();
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

});
