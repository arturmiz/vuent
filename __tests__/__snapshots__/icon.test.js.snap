// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon by default renders correctly 1`] = `<span class="vnt-icon"><!----></span>`;

exports[`Icon given icon's key renders correctly 1`] = `<span class="vnt-icon" name="settings"><!----></span>`;
